eth/provider/provider.go

```diff
@@ -38,6 +38,27 @@ type PolarisProvider struct {
 
 // NewPolarisProvider creates a new `PolarisEVM` instance for use on an underlying blockchain.
 func NewPolarisProvider(
+	configPath string,
+	dataDir string,
+	host core.PolarisHostChain,
+	logHandler log.Handler,
+) *PolarisProvider {
+	// Load the config file.
+	cfg, err := LoadConfigFromFilePath(configPath)
+	if err != nil {
+		// todo: this is hood.
+		cfg = DefaultConfig()
+	}
+
+	// set the data dir
+	cfg.NodeConfig.DataDir = dataDir
```

Comment:
you don't need this anymore, since you can just put the dataDir in the config?

Reply:
I don't fully understand, but if I don't put this, it will set DataDir as "/" then we have the folder permission issue